Gems/Atom/Feature/Common/Code/Source/PostProcessing/ChromaticAberrationPass.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <PostProcessing/ChromaticAberrationPass.h>
+#include <PostProcess/PostProcessFeatureProcessor.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Scene.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<ChromaticAberrationPass> ChromaticAberrationPass::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<ChromaticAberrationPass> pass = aznew ChromaticAberrationPass(descriptor);
+            return AZStd::move(pass);
+        }
+
+        ChromaticAberrationPass::ChromaticAberrationPass(const RPI::PassDescriptor& descriptor)
+            : RPI::ComputePass(descriptor)
+        {
+        }
+
+        bool ChromaticAberrationPass::IsEnabled() const
+        {
+            if (!ComputePass::IsEnabled())
+            {
+                return false;
+            }
+            const RPI::Scene* scene = GetScene();
+            if (!scene)
+            {
+                return false;
+            }
+            PostProcessFeatureProcessor* fp = scene->GetFeatureProcessor<PostProcessFeatureProcessor>();
+            const RPI::ViewPtr view = GetRenderPipeline()->GetDefaultView();
+            if (!fp)
+            {
+                return true;
+            }
+            PostProcessSettings* postProcessSettings = fp->GetLevelSettingsFromView(view);
+            if (!postProcessSettings)
+            {
+                return true;
+            }
+            const ChromaticAberrationSettings* chromaticAberrationSettings = postProcessSettings->GetChromaticAberrationSettings();
+            if (!chromaticAberrationSettings)
+            {
+                return true;
+            }
+            return chromaticAberrationSettings->GetEnabled();
+        }
+
+        void ChromaticAberrationPass::FrameBeginInternal(FramePrepareParams params)
+        {
+            // Must match the struct in .azsl
+            struct Constants
+            {
+                AZStd::array<u32, 2> m_outputSize;
+                AZStd::array<float, 2> m_outputCenter;
+                float m_strength = ChromaticAberration::DefaultStrength;
+                float m_blend = ChromaticAberration::DefaultBlend;
+                float m_blendRemainder = 1 - m_blend;
+            } constants{};
+
+            RPI::Scene* scene = GetScene();
+            PostProcessFeatureProcessor* fp = scene->GetFeatureProcessor<PostProcessFeatureProcessor>();
+            RPI::ViewPtr view = scene->GetDefaultRenderPipeline()->GetDefaultView();
```

Comment:
Drop in the scope below

Reply:
This pattern is repeated all over the engine. I have checked about 8 or 9 examples before responding to this and they all place getting the view outside of the block checking the feature processor. Perhaps it would be better to keep it consistent and move them all in a different PR? Thoughts?

Reply:
Upon reflection, I think this change should be made and an issue raised for the other ppfx. The change will be in the next commit.